build/pluginctl/logs.go

```diff
@@ -0,0 +1,160 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+const (
+	logsPerPage     = 100 // logsPerPage is the number of log entries to fetch per API call
+	timeStampFormat = "2006-01-02 15:04:05.000 Z07:00"
+)
+
+// logs fetches the latest 500 log entries from Mattermost,
+// and prints only the ones related to the plugin to stdout.
+func logs(ctx context.Context, client *model.Client4, pluginID string) error {
+	for i := 5 - 1; i >= 0; i-- {
```

Comment:
Looping here to get `5 * logsPerPage` feels like it introduces the problem of overlapping pages unnecessarily (because new logs are getting written in real-time). Can we simplify and just get all 500 in one pass?

Reply:
Good point on the overlapping pages. Updated the code to fetch 500 entries in one go via 9214379de7d892643cfd3739036c64eec0700e3e.